<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Part 2: Population genetics: the forces that shape genetic variation | _main.knit</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Part 2: Population genetics: the forces that shape genetic variation | _main.knit" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Part 2: Population genetics: the forces that shape genetic variation | _main.knit" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Notebook for HG</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><i class="fa fa-check"></i><b>2</b> Part 2: Population genetics: the forces that shape genetic variation</a>
<ul>
<li class="chapter" data-level="2.1" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#genetic-drift-what-happens-to-alleles-over-time"><i class="fa fa-check"></i><b>2.1</b> 2.1 Genetic Drift: What happens to alleles over time?</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-wright-fisher-wf-model-of-genetic-drift"><i class="fa fa-check"></i><b>2.1.1</b> The Wright-Fisher (WF) model of genetic drift</a></li>
<li class="chapter" data-level="2.1.2" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#binomial-sampling-over-successive-generations-produces-genetic-drift"><i class="fa fa-check"></i><b>2.1.2</b> Binomial sampling over successive generations produces genetic drift</a></li>
<li class="chapter" data-level="2.1.3" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#mutation-and-drift"><i class="fa fa-check"></i><b>2.1.3</b> mutation and drift</a></li>
<li class="chapter" data-level="2.1.4" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#mutation-drift-and-the-amount-of-genetic-variation"><i class="fa fa-check"></i><b>2.1.4</b> mutation, drift and the amount of genetic variation</a></li>
<li class="chapter" data-level="2.1.5" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#effective-population-size"><i class="fa fa-check"></i><b>2.1.5</b> effective population size</a></li>
<li class="chapter" data-level="2.1.6" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-wf-model-with-haplotypes"><i class="fa fa-check"></i><b>2.1.6</b> the WF model with haplotypes</a></li>
<li class="chapter" data-level="2.1.7" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#wf-simulation-of-haplotype-variation"><i class="fa fa-check"></i><b>2.1.7</b> WF simulation of haplotype variation</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#more-on-genetic-drift-the-coalescent"><i class="fa fa-check"></i><b>2.2</b> 2.2 More on genetic drift: The coalescent</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#a-short-history"><i class="fa fa-check"></i><b>2.2.1</b> A short history</a></li>
<li class="chapter" data-level="2.2.2" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#inheritance-of-genetic-material-from-a-shared-ancestor"><i class="fa fa-check"></i><b>2.2.2</b> Inheritance of genetic material from a shared ancestor</a></li>
<li class="chapter" data-level="2.2.3" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-coalescent-refers-to-ancient-shared-ancestry-within-populations"><i class="fa fa-check"></i><b>2.2.3</b> The coalescent refers to ancient shared ancestry within populations</a></li>
<li class="chapter" data-level="2.2.4" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#time-to-coalescence"><i class="fa fa-check"></i><b>2.2.4</b> Time to coalescence</a></li>
<li class="chapter" data-level="2.2.5" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#understanding-waiting-time-distributions-the-geometric-distribution"><i class="fa fa-check"></i><b>2.2.5</b> Understanding waiting-time distributions: the geometric distribution</a></li>
<li class="chapter" data-level="2.2.6" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#understanding-waiting-time-distributions-the-exponential-distribution"><i class="fa fa-check"></i><b>2.2.6</b> Understanding waiting-time distributions: the exponential distribution</a></li>
<li class="chapter" data-level="2.2.7" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-time-distribution-for-two-samples"><i class="fa fa-check"></i><b>2.2.7</b> The time distribution for two samples</a></li>
<li class="chapter" data-level="2.2.8" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-coalescent-for-larger-samples"><i class="fa fa-check"></i><b>2.2.8</b> The coalescent for larger samples</a></li>
<li class="chapter" data-level="2.2.9" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#coalescent-with-mutation"><i class="fa fa-check"></i><b>2.2.9</b> Coalescent with mutation</a></li>
<li class="chapter" data-level="2.2.10" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#quantitative-aspects-of-variation-in-the-coalescent"><i class="fa fa-check"></i><b>2.2.10</b> Quantitative aspects of variation in the coalescent</a></li>
<li class="chapter" data-level="2.2.11" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-coalescent-with-population-size-changes"><i class="fa fa-check"></i><b>2.2.11</b> The coalescent with population size changes</a></li>
<li class="chapter" data-level="2.2.12" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-coalescent-and-the-fixation-process"><i class="fa fa-check"></i><b>2.2.12</b> The coalescent and the fixation process</a></li>
<li class="chapter" data-level="2.2.13" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#coalescent-simulation-of-haplotype-variation"><i class="fa fa-check"></i><b>2.2.13</b> Coalescent simulation of haplotype variation</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#linkage-recombination-and-ld"><i class="fa fa-check"></i><b>2.3</b> 2.3 Linkage, recombination, and LD</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#a-first-look-at-haplotype-structure"><i class="fa fa-check"></i><b>2.3.1</b> A first look at haplotype structure</a></li>
<li class="chapter" data-level="2.3.2" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#linkage-generates-haplotype-structure-or-equivalently-ld"><i class="fa fa-check"></i><b>2.3.2</b> Linkage generates haplotype structure (or equivalently, LD)</a></li>
<li class="chapter" data-level="2.3.3" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#recombination"><i class="fa fa-check"></i><b>2.3.3</b> Recombination</a></li>
<li class="chapter" data-level="2.3.4" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#measuring-ld-between-pairs-of-snps"><i class="fa fa-check"></i><b>2.3.4</b> Measuring LD between pairs of SNPs</a></li>
<li class="chapter" data-level="2.3.5" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#strong-recombination-breaks-down-ld"><i class="fa fa-check"></i><b>2.3.5</b> Strong recombination breaks down LD</a></li>
<li class="chapter" data-level="2.3.6" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-coalescent-with-recombination-the-arg"><i class="fa fa-check"></i><b>2.3.6</b> The coalescent with recombination: the ARG</a></li>
<li class="chapter" data-level="2.3.7" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#recombination-and-ld-in-human-data"><i class="fa fa-check"></i><b>2.3.7</b> Recombination and LD in human data</a></li>
<li class="chapter" data-level="2.3.8" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#prdm9-and-the-hotspot-paradox"><i class="fa fa-check"></i><b>2.3.8</b> PRDM9 and the hotspot paradox</a></li>
<li class="chapter" data-level="2.3.9" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#haplotype-copying-models"><i class="fa fa-check"></i><b>2.3.9</b> Haplotype copying models</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#genetic-drift-in-structured-populations"><i class="fa fa-check"></i><b>2.4</b> Genetic drift in structured populations</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#humans-share-a-recent-african-origin"><i class="fa fa-check"></i><b>2.4.1</b> Humans share a recent African origin</a></li>
<li class="chapter" data-level="2.4.2" data-path="part-2-population-genetics-the-forces-that-shape-genetic-variation.html"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#allele-freqeuncy-variation-across-populations"><i class="fa fa-check"></i><b>2.4.2</b> Allele freqeuncy variation across populations</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
Published with <a href="https://github.com/rstudio/bookdown" target="blank">bookdown</a> by <a href="http://zjuwhw.github.io/" target="blank">zjuwhw</a>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="part-2-population-genetics-the-forces-that-shape-genetic-variation" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">2</span> Part 2: Population genetics: the forces that shape genetic variation<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#part-2-population-genetics-the-forces-that-shape-genetic-variation" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>key forces are:</p>
<ul>
<li>mutation</li>
<li>drift</li>
<li>recombination</li>
<li>natural selection</li>
</ul>
<div id="genetic-drift-what-happens-to-alleles-over-time" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> 2.1 Genetic Drift: What happens to alleles over time?<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#genetic-drift-what-happens-to-alleles-over-time" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li><p>two copies of your genome differ 3 million SNPs</p></li>
<li><p>about 70 are new mutations from your parents (one generation)</p></li>
<li><p>most SNPs are inherited from very distant ancestors</p></li>
<li><p>genetic drift: the random changes in allele frequencies over time</p></li>
<li><p>neutral variation assumption</p></li>
<li><p>mutations can be lost, common, or fixed (fixation)</p></li>
<li><p>original allele/ancestral allele</p></li>
<li><p>new allele/derived allele</p></li>
<li><p>a new derived allele starts out with 1 copy, i.e. allele frequency of the derived allele is <span class="math inline">\(p=1/2N\)</span>, with N be the number of individuals and 2 due to diploid</p></li>
<li><p>Pitcairn island example: 50 inhabitants; 29 founders in 1790; the population never exceeded 250</p></li>
</ul>
<div id="the-wright-fisher-wf-model-of-genetic-drift" class="section level3 hasAnchor" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> The Wright-Fisher (WF) model of genetic drift<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-wright-fisher-wf-model-of-genetic-drift" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Assumption</p>
<ul>
<li><p>a population with N individuals (2N alleles)</p></li>
<li><p>discrete generations</p></li>
<li><p>mate at random</p></li>
<li><p>constant population size (N for every generation)</p></li>
<li><p>ignoring on the sexes of parents</p></li>
<li><p>binomial distribution</p></li>
<li><p>binomial sampling</p></li>
</ul>
<p>2N independent draws, the allele frequency is p in the previous generation and p1 in the next generation</p>
<ul>
<li><span class="math inline">\(E(p1)=E(p)\)</span></li>
<li><span class="math inline">\(var(p1) = p(1-p)/2N\)</span></li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">2</span>)</span>
<span id="cb1-2"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb1-2" aria-hidden="true" tabindex="-1"></a>n<span class="ot">=</span><span class="dv">10</span></span>
<span id="cb1-3"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="do">###  the founder population</span></span>
<span id="cb1-5"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb1-5" aria-hidden="true" tabindex="-1"></a>p0 <span class="ot">=</span> <span class="fl">0.55</span></span>
<span id="cb1-6"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb1-6" aria-hidden="true" tabindex="-1"></a>x0 <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="dv">0</span>,<span class="dv">2</span><span class="sc">*</span>n<span class="sc">*</span>(<span class="dv">1</span><span class="sc">-</span>p0)), <span class="fu">rep</span>(<span class="dv">1</span>, <span class="dv">2</span><span class="sc">*</span>n<span class="sc">*</span>p0))</span>
<span id="cb1-7"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="do">### simulate the next generation</span></span>
<span id="cb1-9"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb1-9" aria-hidden="true" tabindex="-1"></a>replicate_num <span class="ot">=</span> <span class="dv">1000</span></span>
<span id="cb1-10"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb1-10" aria-hidden="true" tabindex="-1"></a>ps <span class="ot">=</span> <span class="fu">c</span>()</span>
<span id="cb1-11"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>replicate_num){</span>
<span id="cb1-12"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb1-12" aria-hidden="true" tabindex="-1"></a>x <span class="ot">=</span> <span class="fu">sample</span>(x0,<span class="dv">2</span><span class="sc">*</span>n,<span class="at">replace=</span>T)</span>
<span id="cb1-13"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb1-13" aria-hidden="true" tabindex="-1"></a>ps<span class="ot">=</span><span class="fu">c</span>(ps, <span class="fu">mean</span>(x))</span>
<span id="cb1-14"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb1-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-15"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="do">### plot the change of allele frequency</span></span>
<span id="cb1-17"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(ps); p0</span></code></pre></div>
<pre><code>## [1] 0.5525</code></pre>
<pre><code>## [1] 0.55</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">var</span>(ps); p0<span class="sc">*</span>(<span class="dv">1</span><span class="sc">-</span>p0)<span class="sc">/</span><span class="dv">2</span><span class="sc">/</span>n</span></code></pre></div>
<pre><code>## [1] 0.01184059</code></pre>
<pre><code>## [1] 0.012375</code></pre>
</div>
<div id="binomial-sampling-over-successive-generations-produces-genetic-drift" class="section level3 hasAnchor" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> Binomial sampling over successive generations produces genetic drift<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#binomial-sampling-over-successive-generations-produces-genetic-drift" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>over the course of many generations</p>
<ul>
<li>the result of binomial sampling in one generation gives you the starting point for binomial sampling in the next generation</li>
<li>Markov chain or random walk</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">21</span>)</span>
<span id="cb7-2"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb7-2" aria-hidden="true" tabindex="-1"></a>n<span class="ot">=</span><span class="dv">10</span></span>
<span id="cb7-3"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="do">### the founder population</span></span>
<span id="cb7-5"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb7-5" aria-hidden="true" tabindex="-1"></a>p0 <span class="ot">=</span> <span class="fl">0.55</span></span>
<span id="cb7-6"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb7-6" aria-hidden="true" tabindex="-1"></a>x0 <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="dv">0</span>,<span class="dv">2</span><span class="sc">*</span>n<span class="sc">*</span>(<span class="dv">1</span><span class="sc">-</span>p0)), <span class="fu">rep</span>(<span class="dv">1</span>, <span class="dv">2</span><span class="sc">*</span>n<span class="sc">*</span>p0))</span>
<span id="cb7-7"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb7-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-8"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="do">### simulate the next 120 generations</span></span>
<span id="cb7-9"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb7-9" aria-hidden="true" tabindex="-1"></a>ps <span class="ot">=</span> <span class="fu">c</span>()</span>
<span id="cb7-10"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb7-10" aria-hidden="true" tabindex="-1"></a>x <span class="ot">=</span> x0</span>
<span id="cb7-11"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">120</span>){</span>
<span id="cb7-12"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb7-12" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">=</span> <span class="fu">sample</span>(x,<span class="dv">2</span><span class="sc">*</span>n,<span class="at">replace=</span>T)</span>
<span id="cb7-13"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb7-13" aria-hidden="true" tabindex="-1"></a>    ps<span class="ot">=</span><span class="fu">c</span>(ps, <span class="fu">mean</span>(x))</span>
<span id="cb7-14"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb7-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-15"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb7-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-16"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="do">### plot the change of allele frequency</span></span>
<span id="cb7-17"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb7-17" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">120</span>, <span class="fu">c</span>(p0, ps), <span class="at">type=</span><span class="st">&quot;l&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<p>Eventually, the derived allele will either reach 100% frequency (fixed) or 0% (lost). 0 and 1 are referred to as absorbing states</p>
</div>
<div id="mutation-and-drift" class="section level3 hasAnchor" number="2.1.3">
<h3><span class="header-section-number">2.1.3</span> mutation and drift<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#mutation-and-drift" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>model the drift for a mutation (only one derived allele)</p>
<p><strong>infinite sites</strong> assumption: each mutation creates a new allele that did not exist previously in the population</p>
<ul>
<li><p>most alleles in a population are very rare: every new mutation starts out rare in the population (one copy), and most are quickly lost, while a very small fraction drift up to become common</p></li>
<li><p>the probability that a derived allele currently at frequency <span class="math inline">\(p\)</span> will eventually fix is also <span class="math inline">\(p\)</span>.</p></li>
</ul>
</div>
<div id="mutation-drift-and-the-amount-of-genetic-variation" class="section level3 hasAnchor" number="2.1.4">
<h3><span class="header-section-number">2.1.4</span> mutation, drift and the amount of genetic variation<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#mutation-drift-and-the-amount-of-genetic-variation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>expected heterozygosity</li>
<li>in modern human populations, expected heterozygosity is ~0.5-1 heterozygous sites per kilobase</li>
<li>a balance between two forces: mutation (input new variation) and drift (remove variation)</li>
<li>it’s <span class="math inline">\(4N\mu\)</span> with N being population size and <span class="math inline">\(\mu\)</span> being the mutation rate</li>
</ul>
</div>
<div id="effective-population-size" class="section level3 hasAnchor" number="2.1.5">
<h3><span class="header-section-number">2.1.5</span> effective population size<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#effective-population-size" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>use effective population size (<span class="math inline">\(Ne\)</span>) instead of N to reflect the actual rate of drift</li>
<li>a few males have very many offspring can greatly reduce <span class="math inline">\(Ne\)</span></li>
<li><span class="math inline">\(Ne\)</span> is usually much smaller than a census estimate of the population size</li>
</ul>
<div id="estimating-ne-from-data" class="section level4 hasAnchor" number="2.1.5.1">
<h4><span class="header-section-number">2.1.5.1</span> Estimating <span class="math inline">\(Ne\)</span> from data<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#estimating-ne-from-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Based on the expected heterozygosity of 5e-4 to 1e-3 and the mutation rate of 1.3e-8 in humans, we can see the <span class="math inline">\(Ne \approx 20,000\)</span> with 10,000 being the lower end of the range.</p>
<p>The low number of Ne 10,000 - 20,000 comparing the current world population of ~8 billion is because it’s a type of average overy the last million years and human population was far smaller for most of that time than it is now.</p>
</div>
</div>
<div id="the-wf-model-with-haplotypes" class="section level3 hasAnchor" number="2.1.6">
<h3><span class="header-section-number">2.1.6</span> the WF model with haplotypes<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-wf-model-with-haplotypes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>haplotype: contain multiple variant sites</p>
<p>assume</p>
<ul>
<li>a genomic region of L basepairs</li>
<li>mutations can arise anywhere in the sequence</li>
<li>each haploid sequence in the next generation is drawn at random from the previous generation</li>
</ul>
</div>
<div id="wf-simulation-of-haplotype-variation" class="section level3 hasAnchor" number="2.1.7">
<h3><span class="header-section-number">2.1.7</span> WF simulation of haplotype variation<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#wf-simulation-of-haplotype-variation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Forward simulation</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">9</span>)</span>
<span id="cb8-2"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-2" aria-hidden="true" tabindex="-1"></a>n <span class="ot">=</span> <span class="dv">5</span>; l<span class="ot">=</span><span class="dv">4</span>; mu<span class="ot">=</span><span class="fl">0.01</span></span>
<span id="cb8-3"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-3" aria-hidden="true" tabindex="-1"></a>G <span class="ot">=</span> <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="at">nrow=</span><span class="dv">2</span><span class="sc">*</span>n, <span class="at">ncol=</span>l)</span>
<span id="cb8-4"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-4" aria-hidden="true" tabindex="-1"></a>nucleotides <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>)</span>
<span id="cb8-5"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-5" aria-hidden="true" tabindex="-1"></a>mutate <span class="ot">=</span> <span class="cf">function</span>(x){</span>
<span id="cb8-6"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">sample</span>(nucleotides[nucleotides<span class="sc">!=</span>x],<span class="dv">1</span>))</span>
<span id="cb8-7"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-8"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-8" aria-hidden="true" tabindex="-1"></a>num_generation <span class="ot">=</span> <span class="dv">5</span></span>
<span id="cb8-9"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-9" aria-hidden="true" tabindex="-1"></a>Glist <span class="ot">=</span> <span class="fu">list</span>(<span class="at">G=</span>G)</span>
<span id="cb8-10"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(igeneration <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>num_generation){</span>
<span id="cb8-11"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(irow <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="dv">2</span><span class="sc">*</span>n)){</span>
<span id="cb8-12"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-12" aria-hidden="true" tabindex="-1"></a>        index_mutation <span class="ot">=</span> <span class="fu">rbinom</span>(l, <span class="dv">1</span>, mu)</span>
<span id="cb8-13"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-13" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">sum</span>(index_mutation)<span class="sc">!=</span><span class="dv">0</span>){</span>
<span id="cb8-14"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-14" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span>(icol <span class="cf">in</span> <span class="fu">which</span>(index_mutation<span class="sc">!=</span><span class="dv">0</span>)){</span>
<span id="cb8-15"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-15" aria-hidden="true" tabindex="-1"></a>                G[irow, icol] <span class="ot">=</span> <span class="fu">mutate</span>(G[irow,icol])</span>
<span id="cb8-16"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-16" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb8-17"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-17" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb8-18"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-18" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb8-19"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-19" aria-hidden="true" tabindex="-1"></a>    G <span class="ot">=</span> G[<span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span>(<span class="dv">2</span><span class="sc">*</span>n), <span class="dv">2</span><span class="sc">*</span>n, <span class="at">replace=</span>T),]</span>
<span id="cb8-20"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-20" aria-hidden="true" tabindex="-1"></a>    Glist[igeneration<span class="sc">+</span><span class="dv">1</span>] <span class="ot">=</span> <span class="fu">list</span>(<span class="at">G=</span>G)</span>
<span id="cb8-21"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-22"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb8-22" aria-hidden="true" tabindex="-1"></a>Glist</span></code></pre></div>
<pre><code>## $G
##       [,1] [,2] [,3] [,4]
##  [1,]    0    0    0    0
##  [2,]    0    0    0    0
##  [3,]    0    0    0    0
##  [4,]    0    0    0    0
##  [5,]    0    0    0    0
##  [6,]    0    0    0    0
##  [7,]    0    0    0    0
##  [8,]    0    0    0    0
##  [9,]    0    0    0    0
## [10,]    0    0    0    0
## 
## [[2]]
##       [,1] [,2] [,3] [,4]
##  [1,]    0    1    0    0
##  [2,]    0    0    0    0
##  [3,]    0    0    0    0
##  [4,]    0    1    0    0
##  [5,]    0    0    0    0
##  [6,]    0    0    0    0
##  [7,]    0    0    0    0
##  [8,]    0    0    0    0
##  [9,]    0    0    0    0
## [10,]    0    0    0    0
## 
## [[3]]
##       [,1] [,2] [,3] [,4]
##  [1,]    0    0    0    0
##  [2,]    0    0    0    2
##  [3,]    0    0    0    0
##  [4,]    0    0    0    0
##  [5,]    0    0    0    2
##  [6,]    0    0    0    0
##  [7,]    0    1    0    0
##  [8,]    0    0    0    0
##  [9,]    0    0    0    0
## [10,]    0    1    0    0
## 
## [[4]]
##       [,1] [,2] [,3] [,4]
##  [1,]    0    0    0    0
##  [2,]    0    1    0    0
##  [3,]    0    1    0    0
##  [4,]    0    0    0    0
##  [5,]    0    0    0    0
##  [6,]    0    0    0    0
##  [7,]    0    0    0    2
##  [8,]    0    0    0    0
##  [9,]    0    0    0    0
## [10,]    0    0    0    0
## 
## [[5]]
##       [,1] [,2] [,3] [,4]
##  [1,]    0    0    0    0
##  [2,]    0    0    0    2
##  [3,]    0    1    0    0
##  [4,]    0    0    0    0
##  [5,]    0    0    0    0
##  [6,]    0    0    0    0
##  [7,]    0    0    0    0
##  [8,]    0    0    0    0
##  [9,]    0    0    0    0
## [10,]    0    0    0    0
## 
## [[6]]
##       [,1] [,2] [,3] [,4]
##  [1,]    0    0    0    0
##  [2,]    0    0    0    0
##  [3,]    0    1    0    0
##  [4,]    0    1    0    0
##  [5,]    0    0    0    0
##  [6,]    0    0    0    0
##  [7,]    0    0    0    0
##  [8,]    0    0    0    0
##  [9,]    0    0    0    0
## [10,]    0    0    0    0</code></pre>
<div id="possible-improvements" class="section level4 hasAnchor" number="2.1.7.1">
<h4><span class="header-section-number">2.1.7.1</span> Possible improvements<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#possible-improvements" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>just keep track of the distinct haplotypes</li>
<li>don’t store the nonvariable sites</li>
<li>generate the total number of new mutations each generation using a single Poisson random variable with mean <span class="math inline">\(2N \mu L\)</span></li>
</ul>
</div>
<div id="simulation-software" class="section level4 hasAnchor" number="2.1.7.2">
<h4><span class="header-section-number">2.1.7.2</span> simulation software<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#simulation-software" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>forward vs backward simulations: computationally slower, but simple models, esaier to implement and far more flexible.</li>
<li><strong>SLiM</strong></li>
</ul>
</div>
</div>
</div>
<div id="more-on-genetic-drift-the-coalescent" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> 2.2 More on genetic drift: The coalescent<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#more-on-genetic-drift-the-coalescent" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="a-short-history" class="section level3 hasAnchor" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> A short history<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#a-short-history" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Wright-Fisher model during the 1920s</li>
<li>Coalescent in 1982 and 1983: John Kingman, Richard Hudson, and Fumio Tajima</li>
</ul>
</div>
<div id="inheritance-of-genetic-material-from-a-shared-ancestor" class="section level3 hasAnchor" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> Inheritance of genetic material from a shared ancestor<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#inheritance-of-genetic-material-from-a-shared-ancestor" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>coalesced; common ancestor</li>
<li>everyone in a population is related, although coalescence is usually far more ancient</li>
</ul>
</div>
<div id="the-coalescent-refers-to-ancient-shared-ancestry-within-populations" class="section level3 hasAnchor" number="2.2.3">
<h3><span class="header-section-number">2.2.3</span> The coalescent refers to ancient shared ancestry within populations<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-coalescent-refers-to-ancient-shared-ancestry-within-populations" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="measuring-time-forward-or-backward" class="section level4 hasAnchor" number="2.2.3.1">
<h4><span class="header-section-number">2.2.3.1</span> Measuring time forward or backward<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#measuring-time-forward-or-backward" class="anchor-section" aria-label="Anchor link to header"></a></h4>
</div>
<div id="the-genealogy-of-a-sample" class="section level4 hasAnchor" number="2.2.3.2">
<h4><span class="header-section-number">2.2.3.2</span> The genealogy of a sample<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-genealogy-of-a-sample" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>coalescent genealogy or tree</li>
<li>a single common ancestor of the entire sample, known as the most recent common ancestor (MRCA)</li>
</ul>
</div>
</div>
<div id="time-to-coalescence" class="section level3 hasAnchor" number="2.2.4">
<h3><span class="header-section-number">2.2.4</span> Time to coalescence<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#time-to-coalescence" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>any two copies of a locus, each copy has a random parent from 2N chromosomes in the previous generation. So the probability that they descend from the same parent is <span class="math inline">\(1/2N\)</span>, and the probability that they descend from different parent is <span class="math inline">\(1-1/2N\)</span>.</p>
<p>The probability that they descend from different parent after going back t generations:</p>
<p><span class="math display">\[(1-\frac{1}{2N})^t\]</span>
as <span class="math inline">\(1-1/2N &lt; 1\)</span>, the <span class="math inline">\((1-1/2N)^t\)</span> approaches zero with t increasing. This means that if we go far enough back in time we can guarantee that any pair of copies of this locus have a common ancestor.</p>
</div>
<div id="understanding-waiting-time-distributions-the-geometric-distribution" class="section level3 hasAnchor" number="2.2.5">
<h3><span class="header-section-number">2.2.5</span> Understanding waiting-time distributions: the geometric distribution<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#understanding-waiting-time-distributions-the-geometric-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Image a 20-sided die, how many times do I need to roll the die until it lands with the ‘20’ face up?</p>
<p>If we get a ‘20’ on t+1 roll, then we don’t have ‘20’ in the previous t rolls and get ‘20’ on the t+1 roll. So the probability is</p>
<p><span class="math display">\[p*(1-p)^t\]</span></p>
<p>, where p is 1/20 in this case</p>
<p>This function describes the waiting times for events and it is called <strong>geometric distribution</strong></p>
<p>The probability of rolling a 20 within the first t rolls:</p>
<p><span class="math display">\[1-(1-p)^t\]</span></p>
<p>The average waiting time to the first success is 1/p</p>
</div>
<div id="understanding-waiting-time-distributions-the-exponential-distribution" class="section level3 hasAnchor" number="2.2.6">
<h3><span class="header-section-number">2.2.6</span> Understanding waiting-time distributions: the exponential distribution<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#understanding-waiting-time-distributions-the-exponential-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>approximate the geometric with a continuous distribution called the <strong>exponential distribution</strong>. They are virtually equivalent</p>
<p>Example: “how long will it be until the next earthquake on the Stanford campus?” And let <span class="math inline">\(\lambda\)</span> be the rate of earthquakes per day.</p>
<p>The probability that the next earthquake will occur exactly t days from now is (probability density function (PDF))</p>
<p><span class="math display">\[\lambda e^{-\lambda t}\]</span></p>
<p>and the total probability of having an earthquake any time within the next t days is (cumulative distribution function (CDF))</p>
<p><span class="math display">\[1-e^{-\lambda t}\]</span></p>
</div>
<div id="the-time-distribution-for-two-samples" class="section level3 hasAnchor" number="2.2.7">
<h3><span class="header-section-number">2.2.7</span> The time distribution for two samples<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-time-distribution-for-two-samples" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For two copies of a locus, there is a probability 1/2N that the two copies will coalesce each generation. We model it by exponential distribution with <span class="math inline">\(\lambda = 1/2N\)</span></p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb10-1" aria-hidden="true" tabindex="-1"></a>n<span class="ot">=</span><span class="dv">20000</span>;lambda<span class="ot">=</span><span class="dv">1</span><span class="sc">/</span><span class="dv">2</span><span class="sc">/</span>n</span>
<span id="cb10-2"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb10-2" aria-hidden="true" tabindex="-1"></a>ts <span class="ot">=</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">200000</span>,<span class="at">length=</span><span class="dv">100</span>)</span>
<span id="cb10-3"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(ts, <span class="dv">1</span><span class="sc">-</span><span class="fu">exp</span>(<span class="sc">-</span>lambda<span class="sc">*</span>ts), <span class="at">type=</span><span class="st">&quot;l&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;p&quot;</span>, <span class="at">main=</span><span class="st">&quot;probability that two samples coalesce </span><span class="sc">\n</span><span class="st"> within the most recent t generations&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;generations&quot;</span>)</span>
<span id="cb10-4"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v=</span><span class="dv">2</span><span class="sc">*</span>n, <span class="at">col=</span><span class="st">&quot;blue&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<p>So, there is a 50% chance that coalescence occurs within the last 1.4N = 28,000 generations. It’s almost certain that coalescence occurs within 10N = 200,000 generations.</p>
<p>These timescales are really lone in terms of human evolution. Assume 25 years for one generation, 2N generations is 1 million years</p>
</div>
<div id="the-coalescent-for-larger-samples" class="section level3 hasAnchor" number="2.2.8">
<h3><span class="header-section-number">2.2.8</span> The coalescent for larger samples<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-coalescent-for-larger-samples" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>extend from a pair of alleles to <span class="math inline">\(m\)</span> alleles (m/2 individuals)</p>
<p>For m copies back in time, pick two of these lineages random to coalesce into a common ancestor, so there will be m-1 copies. This process repeats until we get down to 2, and then finally to one common ancestor.</p>
<p>Use <span class="math inline">\(T_k\)</span> to be the number of generations when there are <span class="math inline">\(k\)</span> lineages on the tree. So <span class="math inline">\(T_2\)</span> has an exponeital distribution. The expected value of <span class="math inline">\(T_k\)</span> is</p>
<p><span class="math display">\[E(T_k) = \frac{4N}{k(k-1)}=2N*\frac{2}{k(k-1)}\]</span></p>
<p>, as for k lineages, there are <span class="math inline">\(k(k-1)/2\)</span> possible pairs to get a possible coalescent.</p>
<p>So when k=10, the average waiting time for a coalescent is 45-fold shorter, 2N/45 generations.</p>
<p>In other words, the most recent coalescent events - when there are many lineages - occur within a few generations, while the oldest coalescent events can easily take a million years.</p>
<p>How long age was the MRCA of a sample (or even of the entire population)?</p>
<p><span class="math display">\[T_{MRCA(m)}=T_2+T_3+T_4+...+T_{m-1}+T_{m}\]</span></p>
<p><span class="math display">\[E[T_{MRCA(m)}]=\sum_{k=2}^{m}\frac{4N}{k(k-1)}\]</span></p>
<p>As the sample size gets large,</p>
<p><span class="math display">\[\lim\limits_{m \to \infty } E[T_{MRCA(m)}] = 4N\]</span></p>
<p>So the common ancestor for the entire population is 4N generations ago (~2 million years, for humans). on average, half of the total time back to the common ancestor is spent waiting for the last two lineages to coalesce.</p>
<p>The genealogy has both random topology (i.e., branching patterns) and random times. both the topology and coalescent times are random draws from the coalescent process.</p>
</div>
<div id="coalescent-with-mutation" class="section level3 hasAnchor" number="2.2.9">
<h3><span class="header-section-number">2.2.9</span> Coalescent with mutation<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#coalescent-with-mutation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Assume a stretch of L base pairs (L = 5kb, for example) in m samples (without recombination) , <span class="math inline">\(\mu\)</span> is the mutation rate per base pair per generation. <span class="math inline">\(b_i\)</span> is the length of branche i on the tree.</p>
<p>Let <span class="math inline">\(n_i\)</span> be the number of mutations on branch <span class="math inline">\(i\)</span>. The expected number of mutations <span class="math inline">\(n_i\)</span> is</p>
<p><span class="math display">\[E[n_i]=\mu b_i L\]</span></p>
<p>, modeled by the Poisson distribution.</p>
<div id="trees-brances-and-derived-allele-frequencies" class="section level4 hasAnchor" number="2.2.9.1">
<h4><span class="header-section-number">2.2.9.1</span> Trees, brances, and derived allele frequencies<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#trees-brances-and-derived-allele-frequencies" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>a branch is above j samples, then any mutation on that branch will occur exactly j times within the sample</p>
<p>terminal branches; mutations that occur on them are referred to as singletons as they are found in only a single sample</p>
</div>
</div>
<div id="quantitative-aspects-of-variation-in-the-coalescent" class="section level3 hasAnchor" number="2.2.10">
<h3><span class="header-section-number">2.2.10</span> Quantitative aspects of variation in the coalescent<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#quantitative-aspects-of-variation-in-the-coalescent" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Question: how many sequence differences can I expect between two samples, in a region of L basepairs?</p>
<p>Recall, the coalescent time for two samples,</p>
<p><span class="math display">\[T_2 \sim exponential(\lambda=1/2N); E(T_2)=2N\]</span></p>
<p>The mutation rate on both branch on L baseparis is <span class="math inline">\(2\mu L\)</span> per generation</p>
<p>So the total number of differences between the two modern-day samples</p>
<p><span class="math display">\[E(T_2) \times 2\mu L=4N\mu L\]</span></p>
<p>It’s convenient to divide this by L, which gives us the expected number of differences per base pair. That is equivalent to <strong>heterozygosity per site</strong>, H, which has been computed based on WF forward model.</p>
<div id="most-heterozygous-snps-are-very-old" class="section level4 hasAnchor" number="2.2.10.1">
<h4><span class="header-section-number">2.2.10.1</span> most heterozygous SNPs are very old<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#most-heterozygous-snps-are-very-old" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Mentioned before that there are about 3 million heterozygous SNPs for two copies of your genome.</p>
<p>Mentioned before the average time to the common ancestor of two homologous copies is 2N generations (~1 million years=<span class="math inline">\(25*2*20,000\)</span>). A mutation, on average, occurs halfway, which is 500,000 years ago.</p>
<p>Modern humans evolved in sub-Saharan Africa. About 70,000 years ago, some populations started spreading out of Africa into the Middle East.</p>
</div>
<div id="the-number-of-snps-found-in-a-sample" class="section level4 hasAnchor" number="2.2.10.2">
<h4><span class="header-section-number">2.2.10.2</span> The number of SNPs found in a sample<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-number-of-snps-found-in-a-sample" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>expand from a sample of 2 to a sample of <span class="math inline">\(m\)</span></p>
<p>Suppose that a region of <span class="math inline">\(L\)</span> basepairs in <span class="math inline">\(m\)</span> samples, the total branch length (i.e., the sum of all the branch lengths) is</p>
<p><span class="math display">\[\sum_{k=2}^m k \times E(T_k)=\sum_{k=2}^m k \times \frac{4N}{k(k-1)}=\sum_{k=2}^m \frac{4N}{k-1}\]</span></p>
<p>So, the expected number of variable sites (i.e., SNPs) is</p>
<p><span class="math display">\[\sum_{k=2}^m \frac{4N}{k-1} \times \mu \times L=4N\mu L\sum_{k=2}^m \frac{1}{k-1}=4N\mu L\sum_{k=1}^m \frac{1}{k}\]</span></p>
<p>One key point is that as the sample size grows the MRCA time converges to 4N, while the number of segregating sites grows indefinitely at a rate proportional to the log of the sample size, <span class="math inline">\(ln(m)\)</span>.</p>
</div>
<div id="the-site-frequency-spectrum-sfs" class="section level4 hasAnchor" number="2.2.10.3">
<h4><span class="header-section-number">2.2.10.3</span> the site frequency spectrum (SFS)<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-site-frequency-spectrum-sfs" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Let <span class="math inline">\(s_i\)</span> be the number of SNPs at which the drived allel is present exactly <span class="math inline">\(i\)</span> times. For example, <span class="math inline">\(s_1\)</span> gives us the number of singletons, <span class="math inline">\(s_2\)</span> the number of doubletons, and so on.</p>
<p>The total number of SNPs, S, is</p>
<p><span class="math display">\[S=\sum_{i=1}^{m-1} s_i\]</span></p>
<p>The vector of allele frequencies <span class="math inline">\(s_1, s_2,s_3,...\)</span> is referred to as the <strong>site frequency spectrum (SFS)</strong>. Distortion away from SFS can be used to detect some types of natural selection, as well as other depatures from the basic model such as recent population growth.</p>
<p>The expected number of variants with a derived allele frequency <span class="math inline">\(i\)</span> is proportional to <span class="math inline">\(1/i\)</span>:</p>
<p><span class="math display">\[E[s_i]=\frac{1}{i}\times 4N\mu L\]</span></p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb11-1" aria-hidden="true" tabindex="-1"></a>m<span class="ot">=</span><span class="dv">100</span>; fNmuL<span class="ot">=</span><span class="dv">100</span></span>
<span id="cb11-2"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(<span class="dv">1</span><span class="sc">/</span>(<span class="dv">1</span><span class="sc">:</span>m)<span class="sc">*</span>fNmuL, <span class="at">col=</span><span class="st">&quot;blue&quot;</span>, <span class="at">border=</span><span class="cn">NA</span>,</span>
<span id="cb11-3"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb11-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">&quot;frequency of derived variant&quot;</span>,</span>
<span id="cb11-4"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb11-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab=</span><span class="st">&quot;expected number of variants&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<p>For m=100, 55% of the variants are below 10% frequency. And for m=10,000, 76% of variants are at &lt;10% frequency.</p>
<p>One key thing is that most variants are rare. A useful rule of thumb is that allele frequencies are uniform on a log scale.</p>
<p>The WF model gives a different conceptual tool to reach a similar conclusion: every new mutation starts out rare and most are lost quickly, while only a few are lucky enough to drift up to become common.</p>
</div>
</div>
<div id="the-coalescent-with-population-size-changes" class="section level3 hasAnchor" number="2.2.11">
<h3><span class="header-section-number">2.2.11</span> The coalescent with population size changes<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-coalescent-with-population-size-changes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The model above, simplest possible population model: constant size and no population structure; <strong>vanilla coalescent</strong></p>
<p>Two types of changing population size: bottlenecks and population growth</p>
<div id="population-bottlenecks" class="section level4 hasAnchor" number="2.2.11.1">
<h4><span class="header-section-number">2.2.11.1</span> Population bottlenecks<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#population-bottlenecks" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>A bottleneck refers to a reduction in population size. Bottlenecks greatly increase the rate of genetic drift.</p>
<p>Bottlenecks during the spread of populations as they left Africa and colonized the globe during the past ~80k years. This is why non-African populations have less genetic variation than Africans.</p>
<p>The ancestors of modern Jews went through a tight population bottleneck ~1000 years ago.</p>
<p>The rate of coalescence is k(k-1)/4N per generation. When N decreases, the rate will increase. This means the few lineages that predate the bottleneck are likely to have many descendants.</p>
</div>
<div id="population-growth" class="section level4 hasAnchor" number="2.2.11.2">
<h4><span class="header-section-number">2.2.11.2</span> Population growth<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#population-growth" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>human populations: ~1 million in 10,000 BCE to ~8 billion today</p>
<p>A very large population size slows down the rate of coalescence at very recent times, and increased the number of very rare variants.</p>
<p>In a model, the population grew instantaneously to infinite size, 100 generations ago. So there would be no coalescence in these 100 generations and greatly extending the terminal branches.</p>
<p>The longer terminal branches produce many more singleton mutations: <span class="math inline">\(4N \times \mu L\)</span> in vanilla model, but <span class="math inline">\((4N+100m) \times \mu L\)</span>.</p>
<p>A more realistic model is continuous exponential growth.</p>
</div>
<div id="footprints-of-population-history-in-real-data" class="section level4 hasAnchor" number="2.2.11.3">
<h4><span class="header-section-number">2.2.11.3</span> Footprints of population history in real data<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#footprints-of-population-history-in-real-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Tennessen et al. (2012), exome sequencing for 1,100 African-American, and 1,300 European-American.</p>
<p>The authors observed a much higher fraction of rare vriants compared to the null, a direct evidence for rapid recent population growth. Then fit a model of historical population sizes (often called a <strong>demographic model</strong>) that can fit the full SFS data.</p>
</div>
</div>
<div id="the-coalescent-and-the-fixation-process" class="section level3 hasAnchor" number="2.2.12">
<h3><span class="header-section-number">2.2.12</span> The coalescent and the fixation process<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-coalescent-and-the-fixation-process" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="the-probability-of-fixation-for-an-allele-now-at-frequency-p-is-simply-p" class="section level4 hasAnchor" number="2.2.12.1">
<h4><span class="header-section-number">2.2.12.1</span> The probability of fixation for an allele now at frequency p, is simply p<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-probability-of-fixation-for-an-allele-now-at-frequency-p-is-simply-p" class="anchor-section" aria-label="Anchor link to header"></a></h4>
</div>
<div id="the-average-time-to-fixation-for-a-new-mutation-is-4n-generations" class="section level4 hasAnchor" number="2.2.12.2">
<h4><span class="header-section-number">2.2.12.2</span> The average time to fixation for a new mutation is 4N generations<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-average-time-to-fixation-for-a-new-mutation-is-4n-generations" class="anchor-section" aria-label="Anchor link to header"></a></h4>
</div>
</div>
<div id="coalescent-simulation-of-haplotype-variation" class="section level3 hasAnchor" number="2.2.13">
<h3><span class="header-section-number">2.2.13</span> Coalescent simulation of haplotype variation<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#coalescent-simulation-of-haplotype-variation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="do">###Goncalo Abecasis&#39;s code: https://csg.sph.umich.edu/abecasis/class/2003/Practical1.html</span></span>
<span id="cb12-2"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">10</span>)</span>
<span id="cb12-4"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-4" aria-hidden="true" tabindex="-1"></a>m<span class="ot">=</span><span class="dv">5</span>; N<span class="ot">=</span><span class="dv">1000</span>; mu<span class="ot">=</span><span class="fl">0.001</span>; L<span class="ot">=</span><span class="dv">1000</span></span>
<span id="cb12-5"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-6"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-6" aria-hidden="true" tabindex="-1"></a>lengths <span class="ot">=</span> <span class="fu">rep</span>(<span class="dv">0</span>,m)</span>
<span id="cb12-7"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-7" aria-hidden="true" tabindex="-1"></a>parentnodes <span class="ot">=</span> <span class="fu">rep</span>(<span class="dv">0</span>,m)</span>
<span id="cb12-8"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-8" aria-hidden="true" tabindex="-1"></a>childnodes <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb12-9"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-9" aria-hidden="true" tabindex="-1"></a>descendants <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">1</span>, m)</span>
<span id="cb12-10"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-10" aria-hidden="true" tabindex="-1"></a>activenodes <span class="ot">=</span> <span class="dv">1</span><span class="sc">:</span>m</span>
<span id="cb12-11"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-11" aria-hidden="true" tabindex="-1"></a>nextNode <span class="ot">=</span> m<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb12-12"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(k <span class="cf">in</span> m<span class="sc">:</span><span class="dv">2</span>){</span>
<span id="cb12-13"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># sample &quot;time to coalescent&quot; from a exponential distribution</span></span>
<span id="cb12-14"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-14" aria-hidden="true" tabindex="-1"></a>    <span class="co"># with a mean of 4N/k(k-1)</span></span>
<span id="cb12-15"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-15" aria-hidden="true" tabindex="-1"></a>    t <span class="ot">=</span> <span class="fu">rexp</span>(<span class="dv">1</span>, <span class="at">rate=</span>k<span class="sc">*</span>(k<span class="dv">-1</span>)<span class="sc">/</span><span class="dv">4</span><span class="sc">/</span>N)</span>
<span id="cb12-16"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># all current nodes increase in length by time t</span></span>
<span id="cb12-17"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-17" aria-hidden="true" tabindex="-1"></a>    lengths[activenodes] <span class="ot">=</span> lengths[activenodes]<span class="sc">+</span>t</span>
<span id="cb12-18"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># sample two nodes to coalesce</span></span>
<span id="cb12-19"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-19" aria-hidden="true" tabindex="-1"></a>    coalescence <span class="ot">=</span> <span class="fu">sample</span>(activenodes, <span class="dv">2</span>)</span>
<span id="cb12-20"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">&quot;After &quot;</span>, t, <span class="st">&quot; generations: Nodes &quot;</span>, coalescence[<span class="dv">1</span>], <span class="st">&quot; and &quot;</span>, coalescence[<span class="dv">2</span>], <span class="st">&quot; coalesce</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb12-21"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-21" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb12-22"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-22" aria-hidden="true" tabindex="-1"></a>    parentnodes[coalescence]<span class="ot">=</span>nextNode</span>
<span id="cb12-23"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-23" aria-hidden="true" tabindex="-1"></a>    childnodes[[nextNode]]<span class="ot">=</span>coalescence</span>
<span id="cb12-24"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-24" aria-hidden="true" tabindex="-1"></a>    activenodes <span class="ot">=</span> <span class="fu">c</span>(activenodes[<span class="sc">!</span>(activenodes <span class="sc">%in%</span> coalescence)], nextNode)</span>
<span id="cb12-25"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-25" aria-hidden="true" tabindex="-1"></a>    parentnodes[nextNode] <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb12-26"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-26" aria-hidden="true" tabindex="-1"></a>    lengths[nextNode] <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb12-27"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-27" aria-hidden="true" tabindex="-1"></a>    descendants[nextNode] <span class="ot">&lt;-</span> descendants[coalescence[<span class="dv">1</span>]] <span class="sc">+</span> descendants[coalescence[<span class="dv">2</span>]]</span>
<span id="cb12-28"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-28" aria-hidden="true" tabindex="-1"></a>    nextNode <span class="ot">=</span> nextNode<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb12-29"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb12-29" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## After  2.991281  generations: Nodes  1  and  2  coalesce
## After  128.7347  generations: Nodes  6  and  5  coalesce
## After  527.5128  generations: Nodes  4  and  3  coalesce
## After  2173.346  generations: Nodes  7  and  8  coalesce</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb14-1" aria-hidden="true" tabindex="-1"></a>lengths</span></code></pre></div>
<pre><code>## [1]    2.991281    2.991281  659.238850  659.238850  131.726002  128.734721
## [7] 2700.858854 2173.346006    0.000000</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb16-1" aria-hidden="true" tabindex="-1"></a>parentnodes</span></code></pre></div>
<pre><code>## [1] 6 6 8 8 7 7 9 9 0</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb18-1" aria-hidden="true" tabindex="-1"></a>activenodes</span></code></pre></div>
<pre><code>## [1] 9</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb20-1" aria-hidden="true" tabindex="-1"></a>descendants</span></code></pre></div>
<pre><code>## [1] 1 1 1 1 1 2 3 2 5</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb22-1" aria-hidden="true" tabindex="-1"></a>childnodes</span></code></pre></div>
<pre><code>## [[1]]
## NULL
## 
## [[2]]
## NULL
## 
## [[3]]
## NULL
## 
## [[4]]
## NULL
## 
## [[5]]
## NULL
## 
## [[6]]
## [1] 1 2
## 
## [[7]]
## [1] 6 5
## 
## [[8]]
## [1] 4 3
## 
## [[9]]
## [1] 7 8</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb24-1" aria-hidden="true" tabindex="-1"></a>spectrum <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">0</span>, m)</span>
<span id="cb24-2"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(descendants)){</span>
<span id="cb24-3"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb24-3" aria-hidden="true" tabindex="-1"></a>spectrum[descendants[i]] <span class="ot">&lt;-</span> spectrum[descendants[i]]<span class="sc">+</span>lengths[i]</span>
<span id="cb24-4"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb24-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-5"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb24-5" aria-hidden="true" tabindex="-1"></a>spectrum</span></code></pre></div>
<pre><code>## [1] 1456.186 2302.081 2700.859    0.000    0.000</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb26-1" aria-hidden="true" tabindex="-1"></a>s <span class="ot">=</span> <span class="fu">sapply</span>(lengths, <span class="cf">function</span>(x) <span class="fu">rpois</span>(<span class="dv">1</span>, x<span class="sc">*</span>mu<span class="sc">*</span>L))</span>
<span id="cb26-2"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb26-2" aria-hidden="true" tabindex="-1"></a>pos <span class="ot">=</span> <span class="fu">sapply</span>(s, <span class="cf">function</span>(x) <span class="fu">as.integer</span>(<span class="fu">runif</span>(x, <span class="at">min=</span><span class="dv">0</span>, <span class="at">max=</span>L)))</span>
<span id="cb26-3"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb26-3" aria-hidden="true" tabindex="-1"></a>mutation <span class="ot">=</span> s</span>
<span id="cb26-4"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb26-4" aria-hidden="true" tabindex="-1"></a>mutation</span></code></pre></div>
<pre><code>## [1]    2    3  617  684  135  124 2752 2207    0</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">length</span>(mutation)<span class="sc">:</span>(m<span class="sc">+</span><span class="dv">1</span>)){</span>
<span id="cb28-2"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb28-2" aria-hidden="true" tabindex="-1"></a>    mutation[childnodes[[i]]] <span class="ot">=</span> mutation[childnodes[[i]]]<span class="sc">+</span>mutation[i]</span>
<span id="cb28-3"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb28-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(mutation)</span>
<span id="cb28-4"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb28-4" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## [1]    2    3  617  684  135  124 2752 2207    0
## [1]    2    3 2824 2891  135  124 2752 2207    0
## [1]    2    3 2824 2891 2887 2876 2752 2207    0
## [1] 2878 2879 2824 2891 2887 2876 2752 2207    0</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb30-1" aria-hidden="true" tabindex="-1"></a>mutation[<span class="dv">1</span><span class="sc">:</span>m]</span></code></pre></div>
<pre><code>## [1] 2878 2879 2824 2891 2887</code></pre>
<p><a href="https://tskit.dev/msprime/docs/stable/intro.html">msprime software</a></p>
</div>
</div>
<div id="linkage-recombination-and-ld" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> 2.3 Linkage, recombination, and LD<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#linkage-recombination-and-ld" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Linkage disequilibrium (LD); recombination breaks down LD by shuffling genotypes</p>
<div id="a-first-look-at-haplotype-structure" class="section level3 hasAnchor" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> A first look at haplotype structure<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#a-first-look-at-haplotype-structure" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First time, Marty Kreitman, 1983, sequenced the ADH gene in 11 lines of the fly Drosophila melanogaster</p>
<p>Haplotype; particular combinations of alleles at different SNPs frequenctly appear together</p>
</div>
<div id="linkage-generates-haplotype-structure-or-equivalently-ld" class="section level3 hasAnchor" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> Linkage generates haplotype structure (or equivalently, LD)<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#linkage-generates-haplotype-structure-or-equivalently-ld" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Linkage: sites that are close together in the genome are usually inherited together</p>
<p>Completely linked: no recombination</p>
<p>Relationship between the branching structure of the tree and the corresponding haplotypes:</p>
<ul>
<li>mutations occurred on the same branch <em>always</em> appear together</li>
<li>mutations occurred on the adjacent branches <em>usually</em> appear together</li>
</ul>
<p>Without recombination, there are very strong constraints on the possible configurations of the derived alleles across haplotypes.</p>
<p>Perfect phylogeny: haplotypes that are consistent with a single tree</p>
</div>
<div id="recombination" class="section level3 hasAnchor" number="2.3.3">
<h3><span class="header-section-number">2.3.3</span> Recombination<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#recombination" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="a-quick-refresher-on-recombination" class="section level4 hasAnchor" number="2.3.3.1">
<h4><span class="header-section-number">2.3.3.1</span> A quick refresher on recombination<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#a-quick-refresher-on-recombination" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Druing meiosis (减数分裂), the maternal and paternal chromosomes are broken and then joined back together so that chromosomes in the resulting gametes are mixtures of the parental chromosomes, called <strong>recombination</strong>, or <strong>crossover</strong>.</p>
<p>More-or-less randomly across the genome; averagely 26 per sperm and 42 per egg.</p>
</div>
<div id="genetic-distance" class="section level4 hasAnchor" number="2.3.3.2">
<h4><span class="header-section-number">2.3.3.2</span> Genetic distance<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#genetic-distance" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Measures the rate of crossover, between different positions along a chromosome; centiMorgans (cM)</p>
<p>Genetic distance x, between two points on a chromsome to be x cM if the average number of crossovers between those two points is x/100 per meiosis, or the x% probability of a crossover per meiosis in that interval.</p>
<p>Recombination rate: in cM/Mb: 100 times the expected number of crossovers per megabase.</p>
<p>The average recombination rate in the human is about 1.2 cM per MB. In other words, there is about a 1.2% probability of a crossover event per megabase.</p>
<p>Recombination generates new combinations of alleles that would not be possible with complete linkage.</p>
<p>For SNPs that are close together (less than ~0.01-0.1cM or 10-100Kb), linkage is a stronger force than recombination and there tends to be strong haplotype structure. At larger distances (more than 0.1cM), recombination is highly effective at shuffling genotypes, and LD is generally week.</p>
</div>
</div>
<div id="measuring-ld-between-pairs-of-snps" class="section level3 hasAnchor" number="2.3.4">
<h3><span class="header-section-number">2.3.4</span> Measuring LD between pairs of SNPs<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#measuring-ld-between-pairs-of-snps" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Imagine two SNPs: one with A and a alleles, and the other with B and b alleles. There are four possible haplotypes: AB, Ab, aB, and ab.</p>
<p>The mesasure D is denfied by</p>
<p><span class="math display">\[D = p_{AB}-p_A p_B\]</span></p>
<p>, where <span class="math inline">\(p_{AB}\)</span> is the frequency of haplotype AB, and <span class="math inline">\(p_A\)</span> and <span class="math inline">\(p_B\)</span> is the frequncy of allele A at the first SNP and the frequency of allele B at the second SNP. As the sign depends on the labeling of the allels, so people use <span class="math inline">\(|D|\)</span></p>
<p>D can be interpreted as the statistical covariance between alleles at the two SNPs if alleles are labeled 0 and 1.</p>
<p>If genotypes at the two SNPs are independent, D = 0.</p>
<p>The second measure of LD is known as <span class="math inline">\(D&#39;\)</span></p>
<p><span class="math display">\[D&#39;=\frac{D&#39;}{D_{max}}\]</span></p>
<p>, where <span class="math inline">\(D_{max}=min(p_A p_b, p_a p_B)\)</span> when <span class="math inline">\(D&gt;0\)</span>, and <span class="math inline">\(D_{max}=min(p_A p_B, p_a p_b)\)</span> when <span class="math inline">\(D&lt;0\)</span></p>
<p>The third measure of LD is called <span class="math inline">\(r^2\)</span></p>
<p><span class="math display">\[r^2=\frac{D^2}{p_A p_a p_B p_b}\]</span></p>
<p>, which ranges from 0 and 1. <span class="math inline">\(r^2=1\)</span> means a perfect LD and there are just two of the four possible haplotypes: i.e., only AB/ab or Ab/aB</p>
</div>
<div id="strong-recombination-breaks-down-ld" class="section level3 hasAnchor" number="2.3.5">
<h3><span class="header-section-number">2.3.5</span> Strong recombination breaks down LD<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#strong-recombination-breaks-down-ld" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Consider only recombination, but no drift. If the c is the proability of crossover per generation between two SNPs. After one generation,</p>
<p><span class="math display">\[D_1=(1-c)D_0\]</span></p>
<p>, where <span class="math inline">\(D_1\)</span> is the LD D measurement in the next generation and <span class="math inline">\(D_0\)</span> is in previous generation.</p>
<p>So After t generations,</p>
<p><span class="math display">\[D_t=(1-c)^tD_0\]</span></p>
<p>We can see unless the recombination rate is very small, LD decays very quickly.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb32-1" aria-hidden="true" tabindex="-1"></a>D0<span class="ot">=</span><span class="fl">0.25</span></span>
<span id="cb32-2"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">100</span>, D0<span class="sc">*</span>(<span class="dv">1</span><span class="fl">-0.05</span>)<span class="sc">^</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">100</span>), <span class="at">type=</span><span class="st">&quot;l&quot;</span>); <span class="fu">text</span>(<span class="dv">25</span>,<span class="fl">0.1</span>,<span class="st">&quot;c=0.05&quot;</span>)</span>
<span id="cb32-3"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="fu">lines</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">100</span>,D0<span class="sc">*</span>(<span class="dv">1</span><span class="fl">-0.1</span>)<span class="sc">^</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">100</span>), <span class="at">type=</span><span class="st">&quot;l&quot;</span>, <span class="at">col=</span><span class="st">&quot;blue&quot;</span>); <span class="fu">text</span>(<span class="dv">21</span>,<span class="fl">0.05</span>,<span class="st">&quot;c=0.1&quot;</span>, <span class="at">col=</span><span class="st">&quot;blue&quot;</span>)</span>
<span id="cb32-4"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="fu">lines</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">100</span>,D0<span class="sc">*</span>(<span class="dv">1</span><span class="fl">-0.5</span>)<span class="sc">^</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">100</span>), <span class="at">type=</span><span class="st">&quot;l&quot;</span>, <span class="at">col=</span><span class="st">&quot;red&quot;</span>); <span class="fu">text</span>(<span class="dv">10</span>,<span class="fl">0.02</span>,<span class="st">&quot;c=0.5&quot;</span>, <span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<p>To summarize,</p>
<ul>
<li>if there is no recombination, the basic properties of the coalescent genealogy tell us to expect strong LD;</li>
<li>at large distances (for SNPs more than a few cMs apart), recombination rapidly eliminates LD</li>
<li>at intermediate distance scales (say ~1-100 kb), recombination and coalescence comppete against each other.</li>
</ul>
</div>
<div id="the-coalescent-with-recombination-the-arg" class="section level3 hasAnchor" number="2.3.6">
<h3><span class="header-section-number">2.3.6</span> The coalescent with recombination: the ARG<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-coalescent-with-recombination-the-arg" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Incorporate recombination into the coalescent prodcues a more complex structure called an <strong>ancestral recombination graph (ARG)</strong></p>
<p>Image two positions in a sequence. Coalescence joins lineages, but now recombination can split sequences apart so that each side of a breakpoint becomes a separate lineage.</p>
<p>Considering more samples and more recombination events, the full ARG contains a series of so-called “marginal trees” at different positions across the sequence.</p>
<div id="breakdown-of-ld-within-the-arg" class="section level4 hasAnchor" number="2.3.6.1">
<h4><span class="header-section-number">2.3.6.1</span> Breakdown of LD within the ARG<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#breakdown-of-ld-within-the-arg" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Recombination can create mixtures of haplotypes.</p>
</div>
<div id="the-tug-of-war-拔河-between-coalescence-and-recombination" class="section level4 hasAnchor" number="2.3.6.2">
<h4><span class="header-section-number">2.3.6.2</span> The tug-of-war (拔河) between coalescence and recombination<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-tug-of-war-拔河-between-coalescence-and-recombination" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The tree-structure of the coalescent creates haplotype structure, while recombination tends to break it apart.</p>
<p>The outcome of this competition is determined by a compound parameter, <span class="math inline">\(4Nc\)</span>: the ratio of the rate of recombination (c) to the rate of coalescence (1/2N).</p>
</div>
<div id="timescales-in-the-arg" class="section level4 hasAnchor" number="2.3.6.3">
<h4><span class="header-section-number">2.3.6.3</span> Timescales in the ARG<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#timescales-in-the-arg" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>For two samples and a region of length c. The rate of coalescence is 1/2N per generation and recombination occurs at a rate c per generation (in either lineage), so 2c in total. So the probability of at least one recombination before coalescence is</p>
<p><span class="math display">\[\frac{2c}{2c+1/2N}=\frac{4Nc}{4Nc+1}\]</span>
Expand from two samples to k lineages. Coalescence decreases the number of lineages from k to k-1; this occurs at a rate of <span class="math inline">\(k(k-1)/4N\)</span> per generation. Recombination increases the number of linesages from k to k+1; this occurs at a rate of ck, where c is the total recombination rate across the segment of interest.</p>
<p>So the probability that next event is a recombination event is</p>
<p><span class="math display">\[\frac{kc}{kc+k(k-1)/4N}=\frac{4Nc}{4Nc+k-1}\]</span></p>
<ul>
<li>4Nc is the natural parameter to describe the role of recombination in an ARG</li>
<li>in large samples, coalescence predominates at recent timescales (when k is large), while recombination is more effective at scrambling the lineages further back in time (when k is small)</li>
</ul>
<p>To summarize,</p>
<ul>
<li>sites that are close together tend to share the same genealogy, hence SNPs are in high LD</li>
<li>genealogies become less and less correlated with increasing genetic distance, thus reducing LD</li>
<li>the scale of LD depends on the product of N and c (usually wrtten as 4Nc)</li>
<li>in latge samples, the most recen coalescent events occur faster than recombination, so closely related haplotypes can be shared over large recombination distances, even at distances where overall LD is low.</li>
</ul>
</div>
<div id="decay-of-r2-with-distance" class="section level4 hasAnchor" number="2.3.6.4">
<h4><span class="header-section-number">2.3.6.4</span> Decay of <span class="math inline">\(r^2\)</span> with distance<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#decay-of-r2-with-distance" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Within a short distance, the expected <span class="math inline">\(r^2\)</span> decays as a function of distance:</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb33-1" aria-hidden="true" tabindex="-1"></a>d <span class="ot">=</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">80</span></span>
<span id="cb33-2"><a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(d, (<span class="dv">10</span><span class="sc">+</span>d)<span class="sc">/</span>(<span class="dv">22</span><span class="sc">+</span><span class="dv">13</span><span class="sc">*</span>d<span class="sc">+</span>d<span class="sc">^</span><span class="dv">2</span>), <span class="at">type=</span><span class="st">&quot;l&quot;</span>, <span class="at">col=</span><span class="st">&quot;red&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;distances in 4Nc&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;r2&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<p>on average 100kb in humans is around 4Nc=80; this model predicts that LD should decay to be low within around 10-100kb.</p>
</div>
</div>
<div id="recombination-and-ld-in-human-data" class="section level3 hasAnchor" number="2.3.7">
<h3><span class="header-section-number">2.3.7</span> Recombination and LD in human data<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#recombination-and-ld-in-human-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>1990s; Human Genome project;</p>
<p>STRs in families (low-resolution: ~2 per Mb)</p>
<ul>
<li>recombination rates vary from about 1-3 cM/Mb (except for the centromere), the higher near the telomeres</li>
<li>Female rates are generally higher than male rates; 1.6-fold</li>
</ul>
<p>high-resolution SNP data</p>
<ul>
<li>LD structure forms blocks of high LD (so called haplotype blocks)</li>
<li>heatmap with color indicating <span class="math inline">\(|D&#39;|\)</span> pr <span class="math inline">\(r^2\)</span></li>
<li>The blocky structure of the D’ matrices suggests that most recombination is taking place at the boundary points between adjacent blocks</li>
</ul>
<p>Use LD structure to estimate a fine-scale recombination</p>
<p>recombination hotspots: ~30,000 identified, additional weaker hotspots</p>
<p>LD is lowest in African populations due to their larger long-term effective population size</p>
</div>
<div id="prdm9-and-the-hotspot-paradox" class="section level3 hasAnchor" number="2.3.8">
<h3><span class="header-section-number">2.3.8</span> PRDM9 and the hotspot paradox<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#prdm9-and-the-hotspot-paradox" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>2005 paper by Simon Myers and collegues: 7-nucleotide sequence motif</p>
<p><strong>hotspot paradox</strong>: first pointed out by Rosie Redfield and colleagues in a 1997 PNAS paper</p>
<p>Meiosis,</p>
<ul>
<li>homologous chromosomes pair up;
= crossovers initiated by one of the two homologous</li>
<li>double-strand break for initiating chromosome and part of the chromosome is chewed back in both directions around the break</li>
<li>the damaged region is repaired using the other chromosome as a template as gene coversion</li>
</ul>
<p>For two chromsomes, one with the hotspot motif and the other without, it will ends with two chromsomes without motifs. Biased gene conversion. A form of selction to remove hotspots.</p>
<p>Hotspot paradox argues that over time, all hotspots will be eliminated</p>
<p><strong>comparing LD in humans and chimpanzees</strong></p>
<ul>
<li>differing at only ~1.4% of sites</li>
<li>no meaningful overlap of hotspot locations between humans and chimps</li>
</ul>
<p><strong>European-American population</strong>: not everyone uses the same hotspots at the same rates</p>
<p>papers in 2010</p>
<ul>
<li>PRDM9 with a “zinc finger” domain that is responsible for DNA binding</li>
<li>PRDM9 binds to the DNA and recruits additional machinery to initiate double-strand breaks</li>
<li>zinc finger DNA-binding domain is encoded within a minisatellite (or VNTRs) repeat section of the gene</li>
<li>highly variable due to misparing during DNA replication</li>
<li>PRDM9 has dozens of alleles found in humans</li>
<li>different alleles have differnt DNA binding preferences</li>
<li>Chimpanzees also have completely different PRDM9 alleles from humans</li>
<li>most rapidly evolving vertebrate genes</li>
</ul>
</div>
<div id="haplotype-copying-models" class="section level3 hasAnchor" number="2.3.9">
<h3><span class="header-section-number">2.3.9</span> Haplotype copying models<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#haplotype-copying-models" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Landmark 2003 paper, Na Li and Matthew Stephens; haplotype copying model (Li and Stephens, or product of approximate conditionals model)</p>
<p>conditional sampling probability for the “next” haplotype in a sample</p>
<ol style="list-style-type: decimal">
<li>within a sample region of DNA sequence, the next haplotype “copy” one already seen, but not be identical due to mutations</li>
<li>over a larger region, the enxt haplotype with switch due to recombination</li>
<li>the rate of both switches and mutations should decrease with the sample size of reference panel</li>
</ol>
<div id="the-conditional-sampling-probability-for-haplotype-copying" class="section level4 hasAnchor" number="2.3.9.1">
<h4><span class="header-section-number">2.3.9.1</span> The conditional sampling probability for haplotype copying<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#the-conditional-sampling-probability-for-haplotype-copying" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Assume <span class="math inline">\(K\)</span> haplotypes observed and genotype data at <span class="math inline">\(S\)</span> SNPs, for the next haplotype from the fist variant poisition s=1 to the last s=S:</p>
<ul>
<li>determining the allele value: the proability of the allele in the new haplotype is equal the allele at site s in haplotype k is <span class="math inline">\(1-[\theta/(K+\theta)]\)</span>, where <span class="math inline">\(\theta\)</span> reflects the rate of mutations or mismatches</li>
<li>recombination: move from SNP s to s+1, the proability we introduce a recombination event is <span class="math inline">\(1-e^{-4Nc_s/K}\)</span>, wjere <span class="math inline">\(c_s\)</span> is the expected number of crossovers between these two SNPs per generation</li>
<li>increment SNP position. Set s to s+1 (Markov process)</li>
</ul>
<p>Hidden Markov Models (HMMs)</p>
</div>
<div id="phasing-and-imputation" class="section level4 hasAnchor" number="2.3.9.2">
<h4><span class="header-section-number">2.3.9.2</span> Phasing and imputation<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#phasing-and-imputation" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>phasing: infer which heterozygous alleles are from the same homologs</li>
<li>imputation: infer genotypes at SNPs that were not on the genotype array</li>
</ul>
</div>
</div>
</div>
<div id="genetic-drift-in-structured-populations" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Genetic drift in structured populations<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#genetic-drift-in-structured-populations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>nonrandom mating is referred to as <strong>population structure</strong></p>
<p>between continents and geographic regions, nearby ethnic groups, towns or villages.</p>
<div id="humans-share-a-recent-african-origin" class="section level3 hasAnchor" number="2.4.1">
<h3><span class="header-section-number">2.4.1</span> Humans share a recent African origin<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#humans-share-a-recent-african-origin" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Humans are descended from populations in sub-Saharan Africa. Around 80,000 years ago, part of this population spread out of Africa, and eventually colonized most of the world’s land masses.</p>
</div>
<div id="allele-freqeuncy-variation-across-populations" class="section level3 hasAnchor" number="2.4.2">
<h3><span class="header-section-number">2.4.2</span> Allele freqeuncy variation across populations<a href="part-2-population-genetics-the-forces-that-shape-genetic-variation.html#allele-freqeuncy-variation-across-populations" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>alleles and haplotypes to drift independently in different populations. Lead to differences in allele and haplotype frequencies.</p>
<ol style="list-style-type: decimal">
<li>alleles that are common in one population tend to be common everywhere</li>
<li>alleles that are rare are usually restricted to a single population or continent</li>
</ol>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "none"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
